phpBB/phpbb/mention/helper/mention_helper.php

```diff
@@ -0,0 +1,211 @@
+	* @param string   $post_text          post text
+	* @param integer  $start_tag_length   Length of tag [mention].
+	* @param integer  $end_tag_length     Length of tag [\mention].
+	* @param integer  $userid_list        Ids of users tagged in the post.
+	*
+	* @return array Consisiting of new text and array of userid.
+	*/
+	private function get_regex_substituted_text($matches, $post_text, $start_tag_length, $end_tag_length, $userid_list)
+	{
+		$users_already_mapped = array();
+		$regexp_matches_count = count($matches[1]);
+		for ($i = 0; $i < $regexp_matches_count; $i++)
+		{
+			$username_clean = utf8_clean_string($matches[1][$i][0]);
+			$startpos = $matches[1][$i][1] - $start_tag_length;
+			$length = strlen($matches[1][$i][0]);
+			$endpos = $matches[1][$i][1] + $length + $end_tag_length - 1;
+			$userid = $userid_list[$username_clean];
+			$add_url_tag = '[url=' . generate_board_url() . '/memberlist.php?mode=viewprofile&u=' . $userid . ']' . $username_clean . '[/url]';
+			if ($i == 0)
+			{
+				$new_post_text = substr($post_text, 0, $startpos);
+				$new_post_text = $new_post_text . $add_url_tag;
+				$prev_end_pos = $endpos + 1;
+			}
+			else
+			{
+				$strip_mention = substr($post_text, $prev_end_pos, $startpos - $prev_end_pos);
+				$new_post_text = $new_post_text . $strip_mention;
+				$new_post_text = $new_post_text . $add_url_tag;
+				$prev_end_pos = $endpos + 1;
+			}
+			if (!in_array($userid_list[$username_clean], $users_already_mapped, true))
+			{
+				$users_already_mapped[] = $userid_list[$username_clean];
+			}
+		}
+		return array('post_text' => $new_post_text, 'users_mapped' => $users_already_mapped) ;
+	}
+
+	/**
+	* Centralized function for calling other functions to extract users, sustitute their
+	* profile links and return the new text back to posting.php.
+	*
+	* @param string   $post_text              post text
+	*
+	* @return array Array of responder data
+	*/
+	public function get_mentioned_users($post_text)
+	{
+		$regular_expression_match = '#\[mention\](.*?)\[/mention\]#';
+		$matches = false;
+		$matches = $this->get_regex_match($regular_expression_match, $post_text);
+		if (count($matches[1]) > 0)
+		{
+			$start_tag_length = strlen('[mention]');
+			$end_tag_length = strlen('[\mention]');
+			$user_list = array();
+			$userid_list = array();
+			$user_list = $this->get_user_list($matches);
+			if (count($user_list) > 0)
+			{
+				$temp_notif_type_object = $this->notification_manager->get_item_type_class('notification.type.mention');
+				$userid_list = $temp_notif_type_object->find_users_for_notification($this->db, $user_list);
+				if (count($userid_list) > 0)
+				{
+					$new_post_data = $this->get_regex_substituted_text($matches, $post_text, $start_tag_length, $end_tag_length, $userid_list);
+					return array('new_post_text' => $new_post_data['post_text'], 'users_mentioned' => $new_post_data['users_mapped'], 'notif_type_object' => $temp_notif_type_object);
```

Comment:
Code consistency, in some places you use the short array syntax, in other places not. Please make your code consistent.